SteamList/Controllers/GamesViewController.swift

```diff
@@ -53,8 +51,25 @@ extension GamesViewController: UITableViewDataSource {
     }
     
     
+     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+         var gameObj = MockData.games[indexPath.row]
+         guard let gameDetailsViewController = self.storyboard?.instantiateViewController(identifier: "GameDetailsViewController",
+        creator: {coder -> GameDetailsViewController? in
+             let cell = tableView.cellForRow(at: indexPath)  as! GameTableViewCell
+             if cell.cellIcon.imageView?.image == UIImage(systemName: "star") {
+                 gameObj.isFavorite = false
+             } else {
+                 gameObj.isFavorite = true
+             }
```

Comment:
What this block of code doing here?
Also, can you move navigation process to separate method?

Reply:
_What this block of code doing here?_ 
could you explain please what does this mean

Reply:
done